Psychtoolbox/PsychDemos/DrawFormattedText2Demo.m

```diff
@@ -0,0 +1,138 @@
+    [~,~,bbox]=DrawFormattedText2(cache,'sx',-300,'transform',{'flip',1});
+    Screen('FrameRect', w, [0 0 255], bbox);
+    [~,~,bbox]=DrawFormattedText2(cache,'sy', 300,'transform',{'rotate',90});
+    Screen('FrameRect', w, [0 255 255], bbox);
+    [~,~,bbox]=DrawFormattedText2(cache,'sy',-300,'transform',{'scale',[2 1]});
+    Screen('FrameRect', w, [255 0 255], bbox);
+    % when called with more alignment inputs, the bounding box of the text
+    % is repositioned:
+    rect = [100 100 350 250];
+    Screen('FrameRect', w, 0, rect);
+    % put text in top left of this box
+    DrawFormattedText2(cache,'sx',0,'sy',0,'xalign','left','yalign','top','winRect',rect);
+    % make another box and put text next to bottom right of this box,
+    % vspace differenty
+    rect = wrect([3:4 3:4])-[650 350 300 100];
+    Screen('FrameRect', w, [255 255 0], rect);
+    DrawFormattedText2('simple\ntest\ntext','win',w,'sx','right','sy','bottom','xalign','left','yalign','bottom','winRect',rect,'vSpacing',1.5);
+    
+    Screen('Flip',w);
+    KbStrokeWait;
+    
+    
+    % draw some text, then draw it again in exact same location but rotated
+    % 180 degrees. should have exact same bounding box
+    [~,~,bbox1,c]=DrawFormattedText2('<size=40>c<i>r<size>a<i>p&\n<size=120>ajX\n<size=60>t<font=comic sans>estddd<color=ff0000>ddda<i>d','win',wpnt,'sx','center','sy',1500,'xalign','left','baseColor',0);
+    [~,~,bbox2]  =DrawFormattedText2(c,'transform',{'rotate',180});
+    Screen('FrameRect',wpnt,[255 0 0 128],bbox1,3);
+    Screen('FrameRect',wpnt,[0 255 0 128],bbox2,3);
+    fprintf('Both bounding boxes exactly the same? %d\n',isequal(bbox1,bbox2));
+    
+    Screen('Flip',w);
+    KbStrokeWait;
+    
+    
+    % Read a text file:
+    fd = fopen([PsychtoolboxRoot 'Contents.m'], 'rt');
+    if fd==-1
+        error('Could not open Contents.m file in PTB root folder!');
+    end
+    t = fread(fd,inf,'*char').';
+    fclose(fd);
+    % flop it all on the screen
+    DrawFormattedText2(t,'win',wpnt,'sx','center','xalign','center','sy',150,'baseColor',0);
+    
+    Screen('Flip',w);
+    KbStrokeWait;
+    
+    % now scroll it over the screen. first pregenerate whole text at middle
+    % of screen, and get cache only
+    [~,~,bbox,cache]=DrawFormattedText2(t,'win',wpnt,'sx','center','xalign','center','sy','center','baseColor',0,'cacheOnly',true);
+    % scroll over screen in 5 seconds
+    frate = Screen('FrameRate',screenNumber);
+    textHeight = RectHeight(bbox);
+    step = textHeight/frate/5;
+    
+    tp=zeros(1, frate*5);
+    for p=1:frate*5
+        [~,~,bbox]=DrawFormattedText2(cache,'sy',-p*step);
+        Screen('FrameRect', w, [255 0 0 128], bbox,2);
+
+        tp(p) = Screen('Flip', w);
+
+        if KbCheck
+            break;
+        end
+    end
+    tp = tp(1:p);
+    fprintf('\nAverage redraw duration for scrolling in msecs: %f\n', 1000 * mean(diff(tp)));
+    
+    % End of demo, close window:
+    sca;
+catch me
```

Comment:
I have tried searching for documentation of octave 3, but cannot find this info. Do you happen to have an octave 3 running where you could quickly check this? If not, I'll install it on some old machine here